src/controllers/__tests__/user.test.js

```diff
@@ -0,0 +1,194 @@
+    connectToMongo();
+    await clearDatabase();
+  });
+
+  afterAll(async () => {
+    await closeDatabase();
+    server.close();
+  });
+
+  describe('GET /api/users/', () => {
+    test('Should response with an error message when there are no users', async () => {
+      const response = await request(server).get(`/api/users/`);
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe('There are no users at the moment!');
+    });
+
+    test('Should send the all the users from database in the response', async () => {
+      await request(server)
+        .post('/api/auth/signup')
+        .set('Content-Type', 'application/json')
+        .send(newValidUser);
+
+      await request(server)
+        .post('/api/auth/signup')
+        .set('Content-Type', 'application/json')
+        .send(newValidUser2);
+
+      const response = await request(server).get(`/api/users/`);
+
+      const responseBody = response.body;
+
+      // eslint-disable-next-line no-underscore-dangle
+      userId = responseBody[0]._id;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody.length).toBe(2);
+      expect(responseBody[0]).toMatchObject(expectedValidUser);
+      expect(responseBody[1]).toMatchObject(expectedValidUser2);
+    });
+  });
+
+  describe('GET /api/users/:id', () => {
+    test('Should send the correct user in the response', async () => {
+      const response = await request(server).get(`/api/users/${userId}`);
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(200);
+      expect(responseBody).toMatchObject(expectedValidUser);
+    });
+
+    test('Should response with an error message when requested user ID does not exist', async () => {
+      const response = await request(server).get(
+        `/api/users/${notExistingUserId}`
+      );
+
+      const responseBody = response.body;
+
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe(
+        "The user with the specified ID wasn't found"
+      );
+    });
+
+    test('Should response with an error message when requested user ID is not valid', async () => {
+      const response = await request(server).get(`/api/users/${invalidId}`);
+      const responseBody = response.body;
+      expect(response.header['content-type']).toContain('application/json');
+      expect(response.statusCode).toBe(422);
+      expect(responseBody.message).toBe('Requested user ID is not valid!');
+    });
+  });
+
+  describe('PUT /api/users/:id', () => {
+    test('Should update a user and save it to database', async () => {
```

Comment:
A better test statement would be: `Should update the requested details of the correct user`